js/apps/system/_admin/aardvark/APP/frontend/js/collections/arangoDocument.js

```diff
@@ -32,25 +32,29 @@ window.ArangoDocument = Backbone.Collection.extend({
     });
   },
 
-  addDocument: function (collectionID, key) {
+  addDocument: function (collectionID, key, body) {
     var self = this;
-    self.createTypeDocument(collectionID, key);
+    self.createTypeDocument(collectionID, key, body);
   },
 
-  createTypeEdge: function (collectionID, from, to, key, callback) {
+  createTypeEdge: function (collectionID, from, to, key, body, callback) {
     var newEdge;
 
-    if (key) {
-      newEdge = JSON.stringify({
-        _key: key,
-        _from: from,
-        _to: to
-      });
-    } else {
-      newEdge = JSON.stringify({
-        _from: from,
-        _to: to
-      });
+    try {
+      if (key) {
+        body._key = key;
+        body._from = from;
+        body._to = to;
+        newEdge = JSON.stringify(body);
+      } else {
+        body._from = from;
+        body._to = to;
+        newEdge = JSON.stringify(body);
+      }
+    }
+    catch(x) {
+      body = {};
```

Comment:
does this mean when the user enters an invalid JSON input, we will silently ignore it?

Reply:
the editor itself will print a warning and deny posting. this is the safeguard.